src/WebJobs.Script/Diagnostics/FileTraceWriter.cs

```diff
@@ -100,30 +83,45 @@ public override void Flush()
                 sb.AppendLine(line);
             }
 
+            if (_currentLogFileInfo == null)
+            {
+                // delay create log file
+                SetLogFile();
+            }
+
             // write all lines in a single file operation
-            string contents = sb.ToString();
+            string content = sb.ToString();
             try
             {
-                lock (_syncLock)
-                {
-                    File.AppendAllText(_currentLogFileInfo.FullName, contents);
-                }
+                AppendToFile(_currentLogFileInfo, content);
             }
-            catch (DirectoryNotFoundException)
+            catch (Exception ex) when (ex is DirectoryNotFoundException || ex is FileNotFoundException)
             {
-                // need to handle cases where log file directories might
+                // need to handle cases where log files or directories might
                 // have been deleted from underneath us
-                Directory.CreateDirectory(_logFilePath);
-                lock (_syncLock)
-                {
-                    File.AppendAllText(_currentLogFileInfo.FullName, contents);
-                }
+                SetLogFile();
+                AppendToFile(_currentLogFileInfo, content);
             }
 
+            // check to see if we need to roll to a new log file
             _currentLogFileInfo.Refresh();
-            if (_currentLogFileInfo.Length > MaxLogFileSizeBytes)
+            if (LogFileIsOverSize(_currentLogFileInfo))
             {
-                SetNewLogFile();
+                SetLogFile();
+            }
+        }
+
+        private static void AppendToFile(FileInfo fileInfo, string content)
+        {
+            lock (_syncLock)
+            {
+                var fs = fileInfo.Open(FileMode.Open, FileAccess.Write);
```

Comment:
Should this also be in a `using` statement? The `StreamWriter` will dispose it for us, but it may be possible for it to throw before that `StreamWriter` is created. For example, if `Seek` (on the next line) throws.

Reply:
That's the way I had it, but it generates the below error:

Severity	Code	Description	Project	File	Line	Suppression State
Error	CA2202	Object 'fs' can be disposed more than once in method 'FileTraceWriter.AppendToFile(FileInfo, string)'. To avoid generating a System.ObjectDisposedException you should not call Dispose more than one time on an object.: Lines: 126	WebJobs.Script	C:\code\Git\Azure\azure-webjobs-sdk-script\src\WebJobs.Script\Diagnostics\FileTraceWriter.cs	126	Active

Reply:
I can have the FileStream in a using block and remove the using for the StreamWriter - that doesn't generate the error.